UKHomeOffice/nmsw-ui
# NMSW-324 Passenger routes

# Ticket

NMSW-324

----

## AC



Page | URL | Link type | Link name | Link route | Next page
-- | -- | -- | -- | -- | --
Are there passengers? | /report-voyage/passenger-details | primary submit button | Save and continue | /report-voyage/upload-passenger-details | Yes: upload passenger details
Are there passengers? | /report-voyage/passenger-details | primary submit button | Save and continue | /report-voyage | No (default): draft declarations tasklist
Upload passenger details | /report-voyage/upload-passenger-details | primary submit button | Check for errors | /report-voyage/passenger-details-uploaded | noErrors: Passenger details uploaded
Upload passenger details | /report-voyage/upload-passenger-details | primary submit button | Check for errors | stay on page | Errors:
Passenger details uploaded | /report-voyage/passenger-details-uploaded | primary submit button | Save and continue | /report-voyage | draft declarations tasklist

Prototype
- is ship carrying any passengers page: /mvp-2/adding-passenger-details
- upload passengers page: /mvp-2/add-passenger-details?section3=

----

## To test

- run tests all should pass
- run app
- go to `/report-voyage/passenger-details`
- select `yes`
- click save and continue
- > you should be taken to `/report-voyage/upload-passenger-details`
- click save and continue
- > you should be taken to `/report-voyage/passenger-details-uploaded`
- go back to `/report-voyage/passenger-details`
- select `no`
- click save and continue
- > you should be taken to `/report-voyage/passenger-details-uploaded`

----

## Notes
- no file uploads
- no passing of voyage ids
